read.go

```diff
@@ -534,31 +534,14 @@ func (d *indexData) newBtreeIndex(toc *indexTOC) (btreeIndex, error) {
 
 	bi.bt = bt
 
-	bi.bucketOffsets = createBucketOffsets(toc.ngramText, btreeBucketSize)
+	bi.ngramSec = toc.ngramText
 
 	bi.postingOffsets = toc.postings.offsets
 	bi.postingDataSentinelOffset = toc.postings.data.off + toc.postings.data.sz
 
 	return bi, nil
 }
 
-// Because we insert ngrams into the btree in order, we can easily reconstruct
-// the buckets from the ngramText simpleSection just by knowing the bucketSize.
-// The last item of the returned slice is the sentinel value sec.off + sec.sz.
-func createBucketOffsets(sec simpleSection, bucketSize int) []uint32 {
-	step := uint32((bucketSize / 2) * ngramEncoding)
-
-	offsets := make([]uint32, 0, ((sec.sz-1)/step)+1)
```

Comment:
sz - 1 will overflow for empty contents, sz is unsigned, so I reckon this is the root cause of the bug.